libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -354,8 +386,43 @@ string ABIFunctions::conversionFunction(Type const& _from, Type const& _to)
 			break;
 		}
 		case Type::Category::FixedPoint:
-			solUnimplemented("Fixed point types not implemented.");
+		{
+			FixedPointType const &fromFixedType = dynamic_cast<FixedPointType const &>(_from);
+			if (toCategory == Type::Category::Integer)
+			{
+				body =
+						Whiskers("converted := <cleanInt>(div(<cleanFixed>(value), <divisor>))")
```

Comment:
This should be `sdiv` for signed.

In general, I think it would be better to add a new function that shifts by powers of ten, one for signed, one for unsigned with the amount of powers being part of the name of the function. There is already a pair of functions `shiftLeftFunction` and `shiftRightFunction` - this would basically be the same thing just for powers of ten.

Reply:
I think that's a bit overkill.

Reply:
To be honest, I don't think so. It is really easy to make a mistake here. If there are multiple places to make a mistake, it is even the harder to find. If a mistake affects all uses of the same function, it is easy to spot.

Reply:
@meowingtwurtle overkill prevents bugs that lead to millions of dollars being lost. Just do the overkill.